src/pages/Main/Modals/ModalJoin.jsx

```diff
@@ -0,0 +1,269 @@
+import React, { useState } from 'react';
+import { GrClose } from 'react-icons/gr';
+import './Modals.scss';
+
+export default function ModalJoin({ handleModal, email }) {
+  const [joinUserInfo, setJoinUserInfo] = useState({
+    email: email,
+    username: '',
+    password: '',
+    privateDataPeriod: '',
+    phoneNumber: '',
+    birthday: '',
+    experienceYear: '',
+    agreement: 0,
+    withdraw: '',
+  });
+
+  const [agreements, setAgreements] = useState({
+    agreement1: false,
+    agreement2: false,
+    agreement3: false,
+  });
+
+  const handleOnSubmit = () => {
+    fetch('API주소', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json;charset=utf-8',
+      },
+      body: JSON.stringify(joinUserInfo),
+    })
+      .then(res => res.json())
+      .then(data => {
+        const message = data.message;
+        if (message === 'user is created') {
+          handleModal('login');
+        } else if (message === 'dataSource Error #createUser') {
+          alert('가입 실패 ㅠㅠ');
+        }
+      });
+  };
+
+  const handleInputChange = e => {
+    const { name, value } = e.target;
+    setJoinUserInfo(prev => ({
+      ...prev,
+      [name]: value,
+    }));
+  };
+
+  const handlePrivateDataPeriodChange = n => {
+    setJoinUserInfo(prev => ({
+      ...prev,
+      privateDataPeriod: n,
+    }));
+  };
+
+  const emailPattern = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+  const emailIsValid = emailPattern.test(joinUserInfo.email);
+  const koreanRegex = /^[가-힣]{2,}$/;
+  const usernameIsValid = koreanRegex.test(joinUserInfo.username);
+  const passwordPattern =
+    /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[$@$!%*?&])[A-Za-z\d$@$!%*?&]{8,}/;
+  const passwordIsValid = passwordPattern.test(joinUserInfo.password);
+
+  const handleCheckboxChange = e => {
+    const { id } = e.target;
+    let bln = !agreements[id];
+    setAgreements(prev => ({
+      ...prev,
+      [id]: bln,
+    }));
```

Comment:
- bln이라는 변수명을 봤을 때에는 어떤 값을 가지는지 유추할 수 없을 것 같습니다.
- 재할당되지 않는 변수이기 때문에 const로 선언하시면 됩니다.
- 이 정도는 곧장 객체의 값으로 넣어줘도 될 것 같네요!
```suggestion
    const { id } = e.target;

    setAgreements(prev => ({
      ...prev,
      [id]: !agreements[id],
    }));
```